### PR TITLE
Fix debugger infinite loop (Windows) #7907

### PR DESCRIPTION
**Pull request by xarkes**
This fixes https://github.com/radare/radare2/issues/7907

### CODE REVIEW

**Comment by xarkes** [comment]
Hmm don't merge I think it broke the debugger actually

File: libr/debug/debug.c [hunk 1]
```diff
@@ -1359,7 +1359,8 @@ R_API int r_debug_child_clone(RDebug *dbg) {
 }
 
 R_API bool r_debug_is_dead(RDebug *dbg) {
-	bool is_dead = (dbg->pid == -1 && strncmp (dbg->h->name, "gdb", 3));
+	bool is_dead = (dbg->pid == -1 && strncmp (dbg->h->name, "gdb", 3))
+		|| (dbg->reason.type == R_DEBUG_REASON_DEAD);
```
Comment (xarkes) [inline libr/debug/debug.c]: Adding this line breaks the debugger.

Comment (xarkes) [inline libr/debug/debug.c]: ping @skuater do you know why ? It sounds logic to me that r_debug_is_dead shall return true if dbg->reason.type is R_DEBUG_REASON_DEAD, but it actually totally breaks the debugger.

Comment (skuater) [inline libr/debug/debug.c]: I made a PR fixing this issue, the only problem is reset dbg->reason.type when reload binary.

Ur fix work if u run a fresh r2 debug session, but dont work when r2 reload a debuggee.

**Comment by radare** [comment]
Merged

**Pull request closed without merging**